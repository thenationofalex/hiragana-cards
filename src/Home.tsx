import _ from 'lodash/fp';
import React from 'react';
import {SafeAreaView, ScrollView, Text, View} from 'react-native';

import {Characters, ICharacters} from './Characters';

const Home = () => {
  const title = 'Hiragana Cards';

  const ListofChars = _.map((value: ICharacters) => (
    <View>
      <Text>{value.ro}</Text>
      <Text>{value.jp}</Text>
    </View>
  ))(Characters);

  return (
    <SafeAreaView>
      <View>
        <Text>{title}</Text>
        <ScrollView>{ListofChars}</ScrollView>
      </View>
    </SafeAreaView>
  );
};

export default Home;
