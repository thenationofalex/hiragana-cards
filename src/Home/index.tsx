import _ from 'lodash/fp';
import React, {useState} from 'react';
import {SafeAreaView, StyleSheet, View} from 'react-native';
import {Characters, ICharacters} from '../Characters';

import CharacterTable from '../CharacterTable';
import {CONSTANTS} from '../Constants';
import LargeCard from '../LargeCard';
import Title from '../Title';

const {SPACER} = CONSTANTS;

const Home = () => {
  const [tableInView, setTableInView] = useState(false);
  const [char, setChar] = useState(Characters[0]);

  const setCharacterInView = (value: ICharacters) => {
    setChar(value);
    setTableInView(false);
  };

  const setNextCharacterInView = (random?: boolean) => {
    let currentIdx = _.findIndex((value: ICharacters) =>
      _.isEqual(value.ro, char.ro),
    )(Characters);

    const next = (): ICharacters => {
      const charactersWithoutSpacer = _.remove(
        (value: ICharacters) => value.ro === SPACER,
      )(Characters);

      if (random) {
        // @ts-ignore
        return _.sample(charactersWithoutSpacer);
      }

      return charactersWithoutSpacer[currentIdx + 1];
    };

    if (currentIdx >= 0 && currentIdx < Characters.length - 1) {
      setChar(next());
    } else {
      setChar(Characters[0]);
    }
  };

  return (
    <SafeAreaView style={!tableInView && styles.largeCardContainer}>
      {tableInView ? (
        <View style={styles.home}>
          <Title title={'🍊 Hiragana'} />
          {/* @ts-ignore */}
          <CharacterTable setCharacterInView={setCharacterInView} />
        </View>
      ) : (
        <LargeCard
          char={char}
          setTableInView={setTableInView}
          setNextCharacterInView={setNextCharacterInView}
        />
      )}
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  largeCardContainer: {
    backgroundColor: 'orange',
  },
  home: {
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    height: '100%',
    backgroundColor: 'white',
  },
});

export default Home;
