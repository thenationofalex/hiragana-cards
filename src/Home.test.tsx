import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';

import Home from './Home';

describe('Home', () => {
  it('Renders correctly', () => {
    const app = render(<Home />);
    expect(app).toMatchSnapshot();
  });
});
