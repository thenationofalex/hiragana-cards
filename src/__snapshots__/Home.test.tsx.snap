// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "orange",
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      style={
        Object {
          "backgroundColor": "orange",
          "display": "flex",
          "height": 1334,
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "height": 1067.2,
            "width": 750,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <View
            alignHeight={false}
            alignWidth={false}
            clickable={true}
            flip={false}
            flipHorizontal={false}
            flipVertical={true}
            friction={6}
            onFlipEnd={[Function]}
            onFlipStart={[Function]}
            perspective={1000}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "perspective": 1000,
                  },
                  Object {
                    "rotateX": "0deg",
                  },
                ],
              }
            }
            useNativeDriver={true}
          >
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  Array [
                    false,
                    false,
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "display": "flex",
                    "height": 1067.2,
                    "justifyContent": "center",
                    "width": 750,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                        "fontSize": 375,
                      },
                    ]
                  }
                >
                  „ÅÇ
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            All cards
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Next card
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</RCTSafeAreaView>
`;
